### DISCUSSION TITLE
Do not append to file when size of m_dic is only one and new key is the same with that one in m_dic

### DISCUSSION THREAD

**Discussion by kaitian521** [Tencent/MMKV#1150] [Ideas]
In some circumstances, I store only one key in MMKV. 

For example, I put user's latest edit content to MMKV, giving me a chance to restore them for next startup of My App

```
mmkv.encode("latest_edit_content", "long content");
```

At present, key-value will be appended to the file, then a rewrite will be called when being full

I think maybe there is a very small improvement for this:

```
if (m_dic.size() == 1 && m_dic.find(key) != m_dic.end()) {
     // no appending, just reset m_actualSize, write key-value, then recalculate crc for metaFile
}
```

Thank you !

**Comment by lingol** [comment]
This could work, but only in SINGLE_PROCESS_MODE. In multi-process mode, MMKV relys on the fact that m_actualSize will increase whenever there's update made by others. Checkout `MMKV::checkLoadData()` & `MMKV::partialLoadFromFile()`.

**Reply by kaitian521** [reply]
Thanks for your reminds. So it would work only if `m_isInterProcess` is false. 
But honest to say, maybe `MMKV_SINGLE_PROCESS` is used even more compared with `MMKV_MULTI_PROCESS`.

I can give a pr if possible（Actually my team use MMKV heavily, feel sorry to annoy you many times 😭）

**Reply by lingol** [reply]
You are welcome to make improvements. There's absolutely no reason to feel sorry for making it better.
